% -*- mode: noweb; noweb-default-code-mode: R-mode; -*-
\documentclass[a4paper]{article}

\title{Application Tutorial: OmicKriging}
\author{Keston Aquino-Michaels, Heather E. Wheeler, Vassily V. Trubetskoy and Hae Kyung Im}


\usepackage{a4wide}
\usepackage{hyperref}
\usepackage{Sweave} 

 
\VignetteIndexEntry{Using the OmicKriging Package} 
\VignetteDepends{OmicKriging, SNPRelate} 
\VignetteKeyword{Overview}



\begin{document}
\input{OmicKriging-vignette_1.2-concordance}
\maketitle
Method citation: Wheeler HE, et al. (2013) Poly-Omic Prediction of Complex Traits: OmicKriging. arXiv:1303.1788 \url{http://arxiv.org/abs/1303.1788}
\begin{center}
\line(1,0){250}
\end{center}
To install from CRAN:
\begin{Schunk}
\begin{Sinput}
> install.packages("OmicKriging")
\end{Sinput}
\end{Schunk}
\begin{center}
\line(1,0){250}
\end{center}
Start by loading OmicKriging functions into R:
\begin{Schunk}
\begin{Sinput}
> library(OmicKriging)
\end{Sinput}
\end{Schunk}
Define paths to the genotype (plink binary pedigree format), gene expression, and phenotype data files (paths may differ based on where the files are located). These files will later be passed to upcoming functions:
\begin{Schunk}

\begin{Schunk}
\begin{Sinput}
> library(OmicKriging)
> ## decompress vingnette data
> untar("data.tar.gz", exdir="vignettes/")
> "%&%" <- function(a, b) paste(a, b, sep="")
> gdsFile <- "gdsTemp.gds"
> ok.dir <- "data/"
> bFile <- ok.dir %&% "ig_genotypes"
> expFile <- ok.dir %&% "ig_gene_exon.txt"
> phenoFile <- ok.dir %&% "ig_pheno.txt"
\end{Sinput}
\end{Schunk}

\end{Schunk}
Load the phenotype data into R:
\begin{Schunk}

\begin{Schunk}
\begin{Sinput}
> pheno <- read.table(phenoFile, header = T)
\end{Sinput}
\end{Schunk}

\end{Schunk}
Load a pre-computed GCTA GRM into R (recommended):
\begin{Schunk}

\begin{Schunk}
\begin{Sinput}
> grmMat <- read_GRMBin(bFile)
\end{Sinput}
\end{Schunk}

\end{Schunk}
Alternatively, to compute the GRM in R start by converting the genotype data from plink binary format into GDS format:
\begin{Schunk}

\begin{Schunk}
\begin{Sinput}
> library(SNPRelate) ## this should be removed
> convert_genotype_data(bFile = bFile, gdsFile = gdsFile)
\end{Sinput}
\begin{Soutput}
Start snpgdsBED2GDS ...
	open /home/vasya/coxlab_projects/turbo_krigr/vignettes/data/ig_genotypes.bed in the SNP-major mode
	open /home/vasya/coxlab_projects/turbo_krigr/vignettes/data/ig_genotypes.fam DONE.
	open /home/vasya/coxlab_projects/turbo_krigr/vignettes/data/ig_genotypes.bim DONE.
Mon Dec  2 15:04:03 2013 	store sample id, snp id, position, and chromosome.
	start writing: 99 samples, 43555 SNPs ...
 	Mon Dec  2 15:04:03 2013	0%
 	Mon Dec  2 15:04:03 2013	100%
Mon Dec  2 15:04:03 2013 	Done.
\end{Soutput}
\end{Schunk}

\end{Schunk}
Subsequently, compute a genetic relatedness matrix (GRM) from the GDS file: 
\begin{Schunk}

\begin{Schunk}
\begin{Sinput}
> grmMat <- make_GRM(gdsFile = gdsFile)